relay-general/src/store/transactions/processor.rs

```diff
@@ -461,23 +472,34 @@ fn scrub_span_description(span: &mut Span) -> Result<(), ProcessingAction> {
     }
 
     let mut scrubbed = span.description.clone();
+    let mut did_scrub = false;
 
     if let Some(is_url_like) = span.op.value().map(|op| op.starts_with("http")) {
         if is_url_like && scrub_identifiers(&mut scrubbed)? {
-            let Some(new_desc) = scrubbed.value() else {
-                    return Ok(());
-                };
-            span.data
-                .get_or_insert_with(BTreeMap::new)
-                // We don't care what the cause of scrubbing was, since we assume
-                // that after scrubbing the value is sanitized.
-                .insert(
-                    "description.scrubbed".to_owned(),
-                    Annotated::new(Value::String(new_desc.to_owned())),
-                );
+            did_scrub = true;
         }
     }
 
+    if let Some(is_db_like) = span.op.value().map(|op| op.starts_with("db")) {
+        if is_db_like && scrub_sql_queries(&mut scrubbed)? {
+            did_scrub = true;
+        }
+    }
+
+    if did_scrub {
+        let Some(new_desc) = scrubbed.value() else {
+            return Ok(());
+        };
```

Comment:
There is also no need for early return here, the following code is super short and can be used directly in the `if` statement. 

```suggestion
        if let Some(new_desc) = scrubbed.into_value() {

```